### PR TITLE
[Keyboard] add primus75 keyboard

### PR DESCRIPTION
**Pull request by moyi4681**

### CODE REVIEW

File: keyboards/iLumkb/primus75/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Primus75", 
+    "url": "", 
+    "maintainer": "moyi4681", 
+    "width": 16, 
+    "height": 6.5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/iLumkb/primus75/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/iLumkb/primus75/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_ALL(
```
Comment (drashna) [inline keyboards/iLumkb/primus75/keymaps/default/keymap.c:20]:
```suggestion
  [0] = LAYOUT_all(
```

File: keyboards/iLumkb/primus75/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,              KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,             KC_DEL,
+    KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_DEL,   KC_BSPC,  KC_HOME,
+    KC_TAB,             KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+    KC_CAPS,            KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,  KC_ENT,             KC_PGDN,
+    KC_LSFT,  KC_NUBS,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,            KC_RSFT,  KC_UP,    KC_END,
+    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  KC_RGUI,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
+  ),
+
+  [1] = LAYOUT_ALL(
```
Comment (drashna) [inline keyboards/iLumkb/primus75/keymaps/default/keymap.c:29]:
```suggestion
  [1] = LAYOUT_alll(
```

File: keyboards/iLumkb/primus75/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_ALL(
```
Comment (drashna) [inline keyboards/iLumkb/primus75/keymaps/via/keymap.c:20]:
```suggestion
  [0] = LAYOUT_all(
```

File: keyboards/iLumkb/primus75/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,              KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,             KC_DEL,
+    KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_DEL,   KC_BSPC,  KC_HOME,
+    KC_TAB,             KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+    KC_CAPS,            KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,  KC_ENT,             KC_PGDN,
+    KC_LSFT,  KC_NUBS,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,            KC_RSFT,  KC_UP,    KC_END,
+    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  KC_RGUI,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
+  ),
+
+  [1] = LAYOUT_ALL(
```
Comment (drashna) [inline keyboards/iLumkb/primus75/keymaps/via/keymap.c:29]:
```suggestion
  [1] = LAYOUT_all(
```

File: keyboards/iLumkb/primus75/primus75.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "quantum.h"
+#define LAYOUT_ALL( \
```
Comment (drashna) [inline keyboards/iLumkb/primus75/primus75.h:18]:
```suggestion
#define LAYOUT_all( \
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/iLumkb/primus75/rules.mk [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/iLumkb/primus75/rules.mk:26]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/iLumkb/primus75/readme.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Primus75
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681), [The QMK Community](https://github.com/qmk)
+* Hardware Supported: Primus75 PCB
+* Hardware Availability: iLumkb
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ilumkb/primus75/:default
```
Comment (fauxpark) [inline keyboards/iLumkb/primus75/readme.md:9]:
```suggestion
    make ilumkb/primus75:default
```

File: keyboards/iLumkb/primus75/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  ),
+
+  [1] = LAYOUT_all(
+    _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  RESET,    _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,  _______,  BL_DEC,   BL_TOGG,  BL_INC,   BL_STEP,  _______,  _______,  _______,  _______,            _______,  _______,  _______,
+    _______,  _______,            _______,                      _______,                                _______,  _______,  _______,  _______,  _______,  _______
+  ),
```
Comment (fauxpark) [inline keyboards/iLumkb/primus75/keymaps/via/keymap.c:36]: You need two extra empty layers here.

File: keyboards/iLumkb/primus75/config.h [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x445A
+#define PRODUCT_ID      0x1014
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    moyi4681
+#define PRODUCT         Primus75
+#define DESCRIPTION     Primus75
```
Comment (fauxpark) [inline keyboards/iLumkb/primus75/config.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please also rename the vendor folder to lowercase.

File: keyboards/iLumkb/primus75/keymaps/via/keymap.c (hunk 9 above)
Comment (moyi4681) [reply keyboards/iLumkb/primus75/keymaps/via/keymap.c:36]: already define VIA_EEPROM_LAYOUT_OPTIONS_SIZE 2

Comment (fauxpark) [reply keyboards/iLumkb/primus75/keymaps/via/keymap.c:36]: That doesn't change the number of layers you need.

Comment (moyi4681) [reply keyboards/iLumkb/primus75/keymaps/via/keymap.c:36]: oh, got it, thanks! already add that

File: keyboards/iLumkb/primus75/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  ),
+    
+  [3] = LAYOUT_all(
+    _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,
+    _______,  _______,            _______,                      _______,                                _______,  _______,  _______,  _______,  _______,  _______
+  ),
```
Comment (fauxpark) [inline keyboards/iLumkb/primus75/keymaps/default/keymap.c:55]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/iLumkb/primus75/readme.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681), [The QMK Community](https://github.com/qmk)
+* Hardware Supported: Primus75 PCB
+* Hardware Availability: iLumkb
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ilumkb/primus75:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/iLumkb/primus75/readme.md:11]:
```suggestion
Flashing example for this keyboard:

    make ilumkb/primus75:default:flash

## Accessing Bootloader Mode

To access Bootloader Mode, hold the key at the top left of the keyboard while connecting the USB cable (also erases persistent settings).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — CHANGES_REQUESTED** [review]
The folder path needs to be fully lowercase.

Try running:

```
cd keyboards/
git mv iLumkb/ ilumkb/
```

then commit and push. The second command might have to be `git mv iLumkb/ ilu/ && git mv ilu/ ilumkb/` if you're on Windows.

File: keyboards/ilumkb/primus75/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_all(
+    KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,              KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,             KC_DEL,
+    KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_DEL,   KC_BSPC,  KC_HOME,
+    KC_TAB,             KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+    KC_CAPS,            KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,  KC_ENT,             KC_PGDN,
+    KC_LSFT,  KC_NUBS,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,            KC_RSFT,  KC_UP,    KC_END,
+    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  KC_RGUI,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
```
Comment (noroadsleft) [inline keyboards/ilumkb/primus75/keymaps/default/keymap.c:26]: Neither of the keymaps includes a keycode to access Layer 1, but both keymaps have keycodes on that layer.

Replacing `KC_RGUI` with `MO(1)`.

```suggestion
    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  MO(1),    KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
```

File: keyboards/ilumkb/primus75/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT_all(
+    KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,              KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,             KC_DEL,
+    KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_DEL,   KC_BSPC,  KC_HOME,
+    KC_TAB,             KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+    KC_CAPS,            KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,  KC_ENT,             KC_PGDN,
+    KC_LSFT,  KC_NUBS,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,            KC_RSFT,  KC_UP,    KC_END,
+    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  KC_RGUI,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
```
Comment (noroadsleft) [inline keyboards/ilumkb/primus75/keymaps/via/keymap.c:26]:
```suggestion
    KC_LCTL,  KC_LGUI,            KC_LALT,                      KC_SPC,                                 KC_RALT,  MO(1),    KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
```

File: keyboards/ilumkb/primus75/info.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Primus75", 
+    "url": "", 
+    "maintainer": "moyi4681", 
+    "width": 16, 
+    "height": 6.5,
```
Comment (noroadsleft) [inline keyboards/ilumkb/primus75/info.json:6]:
```suggestion
    "height": 6.25,
```

File: keyboards/ilumkb/primus75/info.json [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
...label":"H", "x":6.75, "y":3.5}, {"label":"J", "x":7.75, "y":3.5}, {"label":"K", "x":8.75, "y":3.5}, {"label":"L", "x":9.75, "y":3.5}, {"label":":", "x":10.75, "y":3.5}, {"label":"\"", "x":11.75, "y":3.5}, {"label":"Enter", "x":12.75, "y":3.5, "w":2.25}, {"label":"Page Down", "x":15, "y":3.5}, {"label":"Shift", "x":0, "y":4.5, "w":1.25}, {"x":1.25, "y":4.5}, {"label":"Z", "x":2.25, "y":4.5}, {"label":"X", "x":3.25, "y":4.5}, {"label":"C", "x":4.25, "y":4.5}, {"label":"V", "x":5.25, "y":4.5}, {"label":"B", "x":6.25, "y":4.5}, {"label":"N", "x":7.25, "y":4.5}, {"label":"M", "x":8.25, "y":4.5}, {"label":"<", "x":9.25, "y":4.5}, {"label":">", "x":10.25, "y":4.5}, {"label":"?", "x":11.25, "y":4.5}, {"label":"Shift", "x":12.25, "y":4.5, "w":1.75}, {"label":"\u2191", "x":14, "y":4.5}, {"label":"End", "x":15, "y":4.5}, {"label":"Ctrl", "x":0, "y":5.5, "w":1.25}, {"label":"Win", "x":1.25, "y":5.5, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.5, "w":1.25}, {"x":3.75, "y":5.5, "w":6.25}, {"label":"Alt", "x":10, "y":5.5}, {"label":"Fn", "x":11, "y":5.5}, {"label":"Ctrl", "x":12, "y":5.5}, {"label":"\u2190", "x":13, "y":5.5}, {"label":"\u2193", "x":14, "y":5.5}, {"label":"\u2192", "x":15, "y":5.5}]
```
Comment (noroadsleft) [inline keyboards/ilumkb/primus75/info.json:9]:
```suggestion
            "layout": [{"label":"Esc", "x":0, "y":0}, {"label":"F1", "x":1.5, "y":0}, {"label":"F2", "x":2.5, "y":0}, {"label":"F3", "x":3.5, "y":0}, {"label":"F4", "x":4.5, "y":0}, {"label":"F5", "x":6, "y":0}, {"label":"F6", "x":7, "y":0}, {"label":"F7", "x":8, "y":0}, {"label":"F8", "x":9, "y":0}, {"label":"F9", "x":10.5, "y":0}, {"label":"F10", "x":11.5, "y":0}, {"label":"F11", "x":12.5, "y":0}, {"label":"F12", "x":13.5, "y":0}, {"label":"PrtSc", "x":15, "y":0}, {"label":"~", "x":0, "y":1.25}, {"label":"!", "x":1, "y":1.25}, {"label":"@", "x":2, "y":1.25}, {"label":"#", "x":3, "y":1.25}, {"label":"$", "x":4, "y":1.25}, {"label":"%", "x":5, "y":1.25}, {"label":"^", "x":6, "y":1.25}, {"label":"&", "x":7, "y":1.25}, {"label":"*", "x":8, "y":1.25}, {"label":"(", "x":9, "y":1.25}, {"label":")", "x":10, "y":1.25}, {"label":"_", "x":11, "y":1.25}, {"label":"+", "x":12, "y":1.25}, {"x":13, "y":1.25}, {"label":"Backspace", "x":14, "y":1.25}, {"label":"Home", "x":15, "y":1.25}, {"label":"Tab", "x":0, "y":2.25, "w":1.5}, {"label":"Q", "x":1.5, "y":2.25}, {"label":"W", "x":2.5, "y":2.25}, {"label":"E", "x":3.5, "y":2.25}, {"label":"R", "x":4.5, "y":2.25}, {"label":"T", "x":5.5, "y":2.25}, {"label":"Y", "x":6.5, "y":2.25}, {"label":"U", "x":7.5, "y":2.25}, {"label":"I", "x":8.5, "y":2.25}, {"label":"O", "x":9.5, "y":2.25}, {"label":"P", "x":10.5, "y":2.25}, {"label":"{", "x":11.5, "y":2.25}, {"label":"}", "x":12.5, "y":2.25}, {"label":"|", "x":13.5, "y":2.25, "w":1.5}, {"label":"Page Up", "x":15, "y":2.25}, {"label":"Caps Lock", "x":0, "y":3.25, "w":1.75}, {"label":"A", "x":1.75, "y":3.25}, {"label":"S", "x":2.75, "y":3.25}, {"label":"D", "x":3.75, "y":3.25}, {"label":"F", "x":4.75, "y":3.25}, {"label":"G", "x":5.75, "y":3.25}, {"label":"H", "x":6.75, "y":3.25}, {"label":"J", "x":7.75, "y":3.25}, {"label":"K", "x":8.75, "y":3.25}, {"label":"L", "x":9.75, "y":3.25}, {"label":":", "x":10.75, "y":3.25}, {"label":"\"", "x":11.75, "y":3.25}, {"label":"Enter", "x":12.75, "y":3.25, "w":2.25}, {"label":"Page Down", "x":15, "y":3.25}, {"label":"Shift", "x":0, "y":4.25, "w":1.25}, {"x":1.25, "y":4.25}, {"label":"Z", "x":2.25, "y":4.25}, {"label":"X", "x":3.25, "y":4.25}, {"label":"C", "x":4.25, "y":4.25}, {"label":"V", "x":5.25, "y":4.25}, {"label":"B", "x":6.25, "y":4.25}, {"label":"N", "x":7.25, "y":4.25}, {"label":"M", "x":8.25, "y":4.25}, {"label":"<", "x":9.25, "y":4.25}, {"label":">", "x":10.25, "y":4.25}, {"label":"?", "x":11.25, "y":4.25}, {"label":"Shift", "x":12.25, "y":4.25, "w":1.75}, {"label":"Up", "x":14, "y":4.25}, {"label":"End", "x":15, "y":4.25}, {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, {"label":"Win", "x":1.25, "y":5.25, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.25, "w":1.25}, {"x":3.75, "y":5.25, "w":6.25}, {"label":"Alt", "x":10, "y":5.25}, {"label":"Fn", "x":11, "y":5.25}, {"label":"Ctrl", "x":12, "y":5.25}, {"label":"Left", "x":13, "y":5.25}, {"label":"Down", "x":14, "y":5.25}, {"label":"Right", "x":15, "y":5.25}]
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**